lib/tasks/admin/publish_deal.rake

```diff
@@ -0,0 +1,7 @@
+namespace :admin do
+  
+  desc 'Publish deal now for 24 hours'
+  task :publish_deal => :environment do
+    Deal.deals_on_publishing_date.update_all(start_at: Time.current, end_at: Time.current + 24.hours)
```

Comment:
always run your code before raising a PR
this won't work and will raise an exception.
Don't raise a PR in hassle with ensuring its working properly